src/Microsoft.Extensions.FileProviders.Physical/PhysicalFileProvider.cs

```diff
@@ -291,5 +272,29 @@ public IChangeToken Watch(string filter)
 
             return _filesWatcher.CreateFileChangeToken(filter);
         }
+
+        private static IDirectoryContents GetEnumerableContents(string fullPath)
+        {
+            return new EnumerableDirectoryContents(() =>
```

Comment:
Couldn't we pass in the `DirectoryInfo` to `EnumerableDirectoryContents` and have it iterate it lazily (instead of passing in the Func?)